### PR TITLE
NIFI-11126 Delete MultiPart files in ListenHTTP after processing

### PR DESCRIPTION
**Pull request by exceptionfactory**
# Summary

[NIFI-11126](https://issues.apache.org/jira/browse/NIFI-11126) Updates `ListenHTTP` to delete MultiPart elements after processing to avoid leaving files in the system temporary directory.

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [X] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [X] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [X] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [X] Pull Request based on current revision of the `main` branch
- [X] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [ ] Build completed using `mvn clean install -P contrib-check`
  - [ ] JDK 8
  - [ ] JDK 11
  - [ ] JDK 17

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/servlets/ListenHTTPServlet.java [hunk 1]
```diff
@@ -276,18 +274,17 @@ private Set<FlowFile> handleMultipartRequest(HttpServletRequest request, Process
...
+        final Collection<Part> requestParts = request.getParts();
+        for (final Part part : requestParts) {
             FlowFile flowFile = session.create();
             try (OutputStream flowFileOutputStream = session.write(flowFile)) {
                 StreamUtils.copy(part.getInputStream(), flowFileOutputStream);
             }
             flowFile = saveRequestDetailsAsAttributes(request, session, foundSubject, foundIssuer, flowFile);
             flowFile = savePartDetailsAsAttributes(session, part, flowFile, i, requestParts.size());
             flowFileSet.add(flowFile);
+            part.delete();
```
Comment (ChrisSamo632) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/servlets/ListenHTTPServlet.java:287]: Would this be better in a `finally` block to ensure cleanup even if there are exceptions thrown elsewhere in this loop/method after the possible point of file creation?

Comment (exceptionfactory) [reply nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/servlets/ListenHTTPServlet.java]: Thanks @ChrisSamo632, that's a good suggestion. I pushed an update that iterates through the Parts in a finally block where the Servlet calls `handleMultipartRequest()`.

**Comment by exceptionfactory** [comment]
It looks like the tests failed on Windows, reviewing...

```
Error:  org.apache.nifi.processors.standard.TestListenHTTP.testMultipartFormDataRequest  Time elapsed: 0.297 s  <<< FAILURE!
org.opentest4j.AssertionFailedError: MultiPart files found in temporary directory [C:\Users\RUNNER~1\AppData\Local\Temp] ==> expected: <0> but was: <3>
```

**Comment by exceptionfactory** [comment]
It turns out that the current `ListenHTTPServlet` was not closing the Part `InputStream`. I corrected the behavior by moving `Part.getInputStream()` to the try-with-resources block and that works as expected.

**Review by ChrisSamo632 — APPROVED** [review]

**Comment by ChrisSamo632** [comment]
LGTM, thanks for the fix @exceptionfactory , will merge 👍

**Pull request closed without merging**